### ISSUE TITLE
[tree view] component is not accessible in Android Talkback

### ISSUE THREAD

**Issue by krv5vignesh** [mui/mui-x#12275]
The Tree View component does not work in Android when used with Talkback enabled. When Talkback (screen reader for Android) is enabled, the nodes in the Tree View are not working.

- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/mui-org/material-ui/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯

All the below list issues occur when Talkback is enabled.

1. When Talkback is enabled, the tree nodes get focused by the swipe right gesture. But when double tapped, it does not expand or collapsed and instead only announces the current state. 
2. Similar issue occurs on node selection as well. With Talkback enabled and when we use the double-tap gesture to select the node, it does not get selected.
3. The name of the node is not announced when a node is focused

## Expected Behavior 🤔

1. While using Talkback the nodes should expand or collapse when the double tap gesture is used.
2. Tree nodes should be selectable when double tag gesture is used
3. The name of the tree node should be announced when it is focused.

## Steps to Reproduce 🕹

Enable Talkback on an Android mobile and try to interact with the Tree nodes give in the [example page](https://material-ui.com/components/tree-view/#accessibility)

## Context 🔦

I am trying to use the Tree view component in Android.

## Your Environment 🌎

<details>
  <summary>`npx @material-ui/envinfo`</summary>
  
```
    System:
    OS: macOS 10.15.7
  Binaries:
    Node: 14.15.5 - /usr/local/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 6.14.11 - /usr/local/bin/npm
  Browsers:
    Chrome: 90.0.4430.93
    Edge: 90.0.818.49
    Firefox: 88.0
    Safari: 14.1
  npmPackages:
    @material-ui/core: ^4.11.4 => 4.11.4 
    @material-ui/icons: ^4.11.2 => 4.11.2 
    @material-ui/lab: ^4.0.0-alpha.58 => 4.0.0-alpha.58 
    @material-ui/styles:  4.11.4 
    @material-ui/system:  4.11.3 
    @material-ui/types:  5.1.0 
    @material-ui/utils:  4.11.2 
    @types/react:  17.0.4 
    react: ^17.0.2 => 17.0.2 
    react-dom: ^17.0.2 => 17.0.2 

```
</details>

**Search keywords**:

**Comment by noraleonte** [comment]
Closing since it's unlikely it will be picked up
